src/psyclone/parse/algorithm.py

```diff
@@ -160,33 +168,60 @@ def __init__(self, api="", invoke_name="invoke", kernel_path="",
 
     def parse(self, alg_filename):
         '''Takes a PSyclone conformant algorithm file as input and outputs a
-        parse tree of the code contained therein and an object containing
-        information about the 'invoke' calls in the algorithm file and any
-        associated kernels within the invoke calls.
+        parse tree of the code contained therein and an object
+        containing information about the 'invoke' calls in the
+        algorithm file and any associated kernels within the invoke
+        calls. If the NEMO API is being used then the parsed code is
+        returned without any additional information about the code.
 
         :param str alg_filename: The file containing the algorithm code.
+
         :returns: 2-tuple consisting of the fparser2 parse tree of the \
-        algorithm code and an object holding details of the algorithm \
-        code and the invokes found within it.
+            algorithm code and an object holding details of the \
+            algorithm code and the invokes found within it, unless it \
+            is the NEMO API, where the first entry of the tuple is \
+            None and the second is the fparser2 parse tree of the \
+            code.
         :rtype: (:py:class:`fparser.two.Fortran2003.Program`, \
-                 :py:class:`psyclone.parse.FileInfo`)
-        :raises ParseError: if a program, module, subroutine or \
-        function is not found in the input file.
+            :py:class:`psyclone.parse.FileInfo`) or (NoneType, \
+            :py:class:`fparser.two.Fortran2003.Program`)
 
         '''
         self._alg_filename = alg_filename
-
         if self._line_length:
             # Make sure the code conforms to the line length limit.
             check_line_length(alg_filename)
-
         alg_parse_tree = parse_fp2(alg_filename)
 
         if self._api == "nemo":
             # For this API we just parse the NEMO code and return the resulting
             # fparser2 AST with None for the Algorithm AST.
             return None, alg_parse_tree
 
+        return alg_parse_tree, self.invoke_info(alg_parse_tree)
+
+    def invoke_info(self, alg_parse_tree):
+        '''Takes a PSyclone conformant fparser2 representation of an algorithm
```

Comment:
I think "Takes an fparser2 representation of a PSyclone-conformant algorithm code..."

Reply:
Yes, that makes more sense.